lab_3_generate_by_ngrams/main.py

```diff
@@ -221,6 +336,23 @@ def generate_next_token(self, sequence: tuple[int, ...]) -> Optional[dict]:
 
         In case of corrupt input arguments, None is returned
         """
+        if not isinstance(sequence, tuple) or len(sequence) == 0:
+            return None
+
+        context = sequence[-(self._n_gram_size-1):]
+
+        if len(context) > len(sequence):
+            return None
+
+        tokens = {}
+        sort_data = dict(sorted(self._n_gram_frequencies.items(), key=lambda x: (
+            x[1], -x[0][0], -x[0][1])))
+
+        for key in sort_data:
```

Comment:
Can you describe what you are doing this?

Reply:
I changed this part a little bit. I iterate over sorted dict. If a key starts with a needed context, I add the last token from the key and frequency to the empty dict